src/main/java/dk/kb/datahandler/util/HarvestTimeUtil.java

```diff
@@ -49,6 +62,75 @@ public static synchronized void updateDatestampForOaiTarget(OaiTargetDto oaiTarg
         updateDatestampForOaiTarget(oaiTargetNameFile, datestamp);        
     }
     
+    /**
+     * Validate day format is of form at yyyy-MM-dd.
+     * Example: 2024-02-09
+     * 
+     * @param day  Validate is of correct format
+     */
+    public static boolean validateDayFormat(String day) {
+    	SimpleDateFormat simpleDateFormat = new SimpleDateFormat(dayPattern,Locale.getDefault());
+        simpleDateFormat.setLenient(false); //day must exist
+    try {    	
+    	   Date date = simpleDateFormat.parse(day);    	   
+    	}
+    	catch(Exception e) {
+    		log.warn("Parsing of day format failed:"+day);
+    	    return false;
+    	}
+        return true;			
+    	    	
+    }
+    
+    /**
+     * 
+     * From a java date format yyyy-MM-dd
+     * 
+     * @param date  Java date object
+     */
+    public static String formatDate2Day(Date date) {
+    	SimpleDateFormat simpleDateFormat = new SimpleDateFormat(dayPattern,Locale.getDefault());    	    	    	
+    	String day = simpleDateFormat.format(date);    	       	    
+        return day;			    	    
+    }
+    
+    
+    /**
+     * Return the next day. Return null if next day will be after tomorrow.  <br>
+     * So if today is 2024-02-07, the last day that will be return is 2024-02-08. Else it will be null.
+     * 
+     * 
+     * For input 2024-02-07 the output will be 2024-02-08  
+     * 
+     * @param day day in format yyyy-MM.dd
+     */
+    public static String getNextDayIfNot2DaysInFuture(String day) throws InvalidArgumentServiceException {
+    	SimpleDateFormat simpleDateFormat = new SimpleDateFormat(dayPattern,Locale.getDefault());
+    	Date date = null;
+    	try{
+    		 date = simpleDateFormat.parse(day);
+    	}
+    	catch(Exception e) {
+    		throw new InvalidArgumentServiceException("Not valid day:"+day);
+    	}
+    	
+    	//Add one day
+    	Calendar nextDayCal = Calendar.getInstance(TimeZone.getDefault(),Locale.getDefault());
```

Comment:
Same argument for `Timezone.getDefault()` as for `Locale.getDefault()`: Don't use it as it depends on the local system setup.

As OAI-PMH operates with Zulu time, `TimeZone.getTimeZone("UTC")` seems fitting.

Reply:
TimeZone.getTimeZone("UTC") yes.